### PR TITLE
OSDOCS-5157:Updates to dynamic plugin extensions and api docs

### PR DESCRIPTION
**Pull request by opayne1**
[OSDOCS-5157](https://issues.redhat.com//browse/OSDOCS-5157):Updates to dynamic plugin extensions and api docs

Version(s):
 4.13

Issue:
https://issues.redhat.com/browse/OSDOCS-5157

Link to docs preview:
extensions: https://57513--docspreview.netlify.app/openshift-enterprise/latest/web_console/dynamic-plug-in/dynamic-plug-ins-reference.html#dynamic-plug-in-sdk-extensions_dynamic-plug-ins-reference
API:https://57513--docspreview.netlify.app/openshift-enterprise/latest/web_console/dynamic-plug-in/dynamic-plug-ins-reference.html#dynamic-plug-in-api_dynamic-plug-ins-reference

QE review:
- [ ] QE has approved this change.

Additional information:
**Note for Reviewer:** 
I am not really sure how to get the red highlight off on some of the code blocks in the API docs other than that this is ready for review. It should be reviewed like the API docs as these should stay true to what the console team has in these files https://github.com/openshift/console/tree/master/frontend/packages/console-dynamic-plugin-sdk/docs

### CODE REVIEW

**Comment by opayne1** [comment]
@spadgett @jhadvig Hi Sam and Jakub, I was looking to update the extensions/api docs for 4.13 and I realized the old PR was never merged. I think we had some [open comments](https://github.com/openshift/openshift-docs/pull/55844/files#r1109790265) in there. I think I have addressed them with this PR.

File: modules/dynamic-plug-in-sdk-extensions.adoc [hunk 1]
```diff
@@ -1025,7 +1110,7 @@ events
 [discrete]
 == `console.topology/adapter/build`
 
-`BuildAdapter` contributes an adapter to adapt element to data that can be used by the `Build` component.
+`BuildAdapter` contributes an adapter to adapt element to data that can be used by the build configuration.
```
[the remark this answers is not in the document]
Comment (jhadvig) [reply modules/dynamic-plug-in-sdk-extensions.adoc:1113]: I've changed it on the master https://github.com/openshift/console/pull/12683

[the remark this answers is not in the document]
Comment (opayne1) [reply modules/dynamic-plug-in-sdk-extensions.adoc]: Thanks @jhadvig I have updated this PR to reflect yours.

**Comment by opayne1** [comment]
@yapei I think I will need a QE review of this PR. It is just updating the docs to match what is in https://github.com/opayne1/console/tree/master/frontend/packages/console-dynamic-plugin-sdk/docs. Thank you!

File: modules/dynamic-plug-in-sdk-extensions.adoc [hunk 2]
```diff
@@ -82,19 +82,23 @@ which returns actions for the given resource model
...
 |===
 |Name |Value Type |Optional |Description
-|`alert` |`string` |no |
+|`alert` |`string` |no | Alert name as defined by `alert.rule.name` property |
 
-|`text` |`string` |no |
+|`text` |`string` |no |  Action text |
 
-|`action` |`CodeRef<(alert: any) => void>` |no |
+|`action` |`CodeRef<(alert: any) => void>` |no | Function to perform side effect |
```
Comment (yapei) [inline modules/dynamic-plug-in-sdk-extensions.adoc:94]: it looks like we have some formatting issue in the preview
Screenshot 2023-03-30 at 2 02 07 PM

File: modules/dynamic-plug-in-sdk-extensions.adoc [hunk 3]
```diff
@@ -982,22 +1025,62 @@ resource page will be rendered.
 component to be rendered when the model matches
 |===
 
+[discrete]
+== `console.storage-provisioner`
```
Comment (yapei) [inline modules/dynamic-plug-in-sdk-extensions.adoc:1029]: The extension is `console.storage-class/provisioner`

File: modules/dynamic-plug-in-sdk-extensions.adoc [hunk 4]
```diff
@@ -982,22 +1025,62 @@ resource page will be rendered.
...
+|`OTHERS`
+|`ProvisionerDetails`
+|yes
+|Other provisioner type
+|===
+
 [discrete]
 == `console.storage-provider`
 
+This extension can be used to contribute a new storage provider to select, when attaching storage and a provider specific component.
```
Comment (yapei) [inline modules/dynamic-plug-in-sdk-extensions.adoc:1047]: also formatting issue for `console.storage-provider` 
Screenshot 2023-03-30 at 2 08 47 PM

File: modules/dynamic-plug-in-sdk-extensions.adoc [hunk 5]
```diff
@@ -1025,14 +1110,14 @@ events
...
-`BuildAdapter` contributes an adapter to adapt element to data that can be used by the `Build` component.
+`BuildAdapter` contributes an adapter to adapt element to data that can be used by the Build component.
 
 [cols=",,,",options="header",]
 |===
 |Name |Value Type |Optional |Description
 |`adapt`
 |`CodeRef<(element: GraphElement) => AdapterDataType<BuildConfigData> | undefined>`
-|no |
+|no |adapter to adapt element to data that can be used by Build component.
```
Comment (yapei) [inline modules/dynamic-plug-in-sdk-extensions.adoc:1120]: formatting issue in the table
Screenshot 2023-03-30 at 2 12 58 PM

File: modules/dynamic-plug-in-sdk-extensions.adoc [hunk 6]
```diff
@@ -1045,7 +1130,7 @@ events
 |Name |Value Type |Optional |Description
 |`adapt`
 |`CodeRef<(element: GraphElement) => NetworkAdapterType | undefined>`
-|no |
+|no |adapter to adapt element to data that can be used by the build configuration.
```
Comment (yapei) [inline modules/dynamic-plug-in-sdk-extensions.adoc:1133]: For extension `console.topology/adapter/network`, the description for `adapt` should be 
```
adapter to adapt element to data that can be used by Networking component.
```

File: modules/dynamic-plug-in-sdk-extensions.adoc [hunk 7]
```diff
@@ -1058,7 +1143,7 @@ events
 |Name |Value Type |Optional |Description
 |`adapt`
 |`CodeRef<(element: GraphElement) => AdapterDataType<PodsAdapterDataType> | undefined>`
-|no |
+|no | adapter to adapt element to data that can be used by Pod component. |
```
Comment (yapei) [inline modules/dynamic-plug-in-sdk-extensions.adoc:1146]: also formatting issue in table
Screenshot 2023-03-30 at 2 14 54 PM

File: modules/dynamic-plug-in-sdk-extensions.adoc [hunk 8]
```diff
@@ -1216,8 +1301,8 @@ Topology Display Filters Extension
 [cols=",,,",options="header",]
 |===
 |Name |Value Type |Optional |Description
-|`getTopologyFilters` |`CodeRef<() => TopologyDisplayOption[]>` |no |
-|`applyDisplayOptions` |`CodeRef<TopologyApplyDisplayOptions>` |no |
+|`getTopologyFilters` |`CodeRef<() => TopologyDisplayOption[]>` |no | Getter for topology filters specific to the extension |
+|`applyDisplayOptions` |`CodeRef<TopologyApplyDisplayOptions>` |no | Function to apply filters to the model |
```
Comment (yapei) [inline modules/dynamic-plug-in-sdk-extensions.adoc:1305]: formatting issue in table
Screenshot 2023-03-30 at 2 16 29 PM

**Comment by jhadvig** [comment]
@opayne1 fixed the remaining discrepancy in https://github.com/openshift/console/pull/12683

File: modules/dynamic-plug-in-api.adoc
Comment (michaelryanpeter) [inline modules/dynamic-plug-in-api.adoc:1]: To fix the issue with the microsoft.github links resolving incorrectly, add the following to  line 5 of this module:
```
:power-bi-url: https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.requestinit.html
```
image

File: modules/dynamic-plug-in-api.adoc [hunk 9]
```diff
@@ -801,23 +801,21 @@ model. In case of failure, the promise gets rejected with HTTP error response.
...
-|`` |options.queryParams - The query parameters to be included in the
+|`options.queryParams` |The query parameters to be included in the
 URL.
 
-|`` |options.requestInit - The fetch init object to use. This can have
-request headers, method, redirect, etc. See more (link:
-https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.requestinit.html)
-
+|`options.requestInit` |The fetch init object to use. This can have
+request headers, method, redirect, etc. See more link:https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.requestinit.html[Interface RequestInit]
```
Comment (michaelryanpeter) [inline modules/dynamic-plug-in-api.adoc:818]:
```suggestion
request headers, method, redirect, etc. See more link:{power-bi-url}[Interface RequestInit]
```

File: modules/dynamic-plug-in-api.adoc [hunk 10]
```diff
@@ -903,21 +901,20 @@ It deletes resources from the cluster, based on the provided model, resource. Th
...
-|`` |options.queryParams - The query parameters to be included in the
+|`options.queryParams` |The query parameters to be included in the
 URL.
 
-|`` |options.requestInit - The fetch init object to use. This can have
-request headers, method, redirect, etc. See more \{@link
-https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.requestinit.html
-}
+|`options.requestInit` |The fetch init object to use. This can have
+request headers, method, redirect, etc. See more link:https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.requestinit.html[Interface Request Init]
```
Comment (michaelryanpeter) [inline modules/dynamic-plug-in-api.adoc:914]:
```suggestion
request headers, method, redirect, etc. See more link:{power-bi-url}[Interface RequestInit]
```

File: modules/dynamic-plug-in-api.adoc [hunk 11]
```diff
@@ -931,15 +928,13 @@ Lists the resources as an array in the cluster, based on provided options. It re
...
-|`` |options.queryParams - The query parameters to be included in the
+|`options.queryParams` |The query parameters to be included in the
 URL and can pass label selector's as well with key "labelSelector".
 
-|`` |options.requestInit - The fetch init object to use. This can have
-request headers, method, redirect, etc. See more \{@link
-https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.requestinit.html
-}
+|`options.requestInit` |The fetch init object to use. This can have
+request headers, method, redirect, etc. See more link:https://microsoft.github.io/PowerBI-JavaScript/interfaces/_node_modules_typedoc_node_modules_typescript_lib_lib_dom_d_.requestinit.html[Interface RequestInit]
```
Comment (michaelryanpeter) [inline modules/dynamic-plug-in-api.adoc:937]:
```suggestion
request headers, method, redirect, etc. See more link:{power-bi-url}[Interface RequestInit]

File: modules/dynamic-plug-in-api.adoc
Comment (michaelryanpeter) [reply modules/dynamic-plug-in-api.adoc:1]: Consider adding a comment explaining the rationale for creating a local attribute for the URL.

**Review by sheriff-rh — COMMENTED** [review]
LGTM, no suggested changes! 

Regarding the red `>`,  I don't have any guidance on how to avoid this. Sometimes it can be the wrong language has been selected (`[source,tsx]`), or the rendering believes it's a dangling symbol.

**Pull request merged**